firmware/common/usb_standard_request.c

```diff
@@ -367,6 +366,35 @@ static usb_request_status_t usb_standard_request_get_status(
 	}
 }
 
+static usb_request_status_t usb_standard_request_clear_feature_setup(
+	usb_endpoint_t* const endpoint)
+{
+	switch (endpoint->setup.value) {
+		case USB_FEATURE_SELECTOR_ENDPOINT_HALT:
+			usb_endpoint_reset_data_toggle(endpoint);
```

Comment:
I think this should be reading the target endpoint from `endpoint->index` instead, otherwise it'll target the control endpoint?

Reply:
Of course! That makes sense.

Reply:
I've changed this to look up the endpoint according to the request index but it's still hanging up in the same way, so I think I must still have it wrong somehow.

Reply:
Oh, I think it's also missing a `schedule_ack` call.

Reply:
Right! That did the trick.